pkg/api/auth_middleware.go

```diff
@@ -83,12 +125,31 @@ func checkSecurityRequirements(r *http.Request, securityRequirements openapi3.Se
 				}
 				user, err = userByAuth(ctx, logger, authenticator, authService, accessKey, secretKey)
 			case "cookie_auth":
-				// validate jwt token from cookie
-				jwtCookie, _ := r.Cookie(JWTCookieName)
-				if jwtCookie == nil {
+				var internalAuthSession *sessions.Session
+				internalAuthSession, _ = sessionStore.Get(r, InternalAuthSessionName)
+				token := ""
+				if internalAuthSession != nil {
+					token, _ = internalAuthSession.Values[TokenSessionKeyName].(string)
+				}
+				if token == "" {
+					migrateFromLegacyCookie(r, w, logger, sessionStore)
+				}
+				internalAuthSession, err = sessionStore.Get(r, InternalAuthSessionName)
```

Comment:
We need to clear the cookie as you did - but reading the session after saving it - I don't think it will work.
When you "save" a cookie - you write a header. When you "read" a cookie, it is from the request header that we already parsed.
I maybe wrong as the code can implement some kind of workaround - just wanted to verify with you how it was implemented.